src/vm/amd64/jithelpers_fastwritebarriers.S

```diff
@@ -244,3 +246,331 @@ PATCH_LABEL JIT_WriteBarrier_SVR64_PatchLabel_CardTable
         mov     byte ptr [rdi + rax], 0FFh
         ret
 LEAF_END_MARKED JIT_WriteBarrier_SVR64, _TEXT
+
+#endif
+
+
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+
```

Comment:
These were written 10+ years ago in quite different world, to optimize for early versions of x64 hardware: 
- There was no address space randomization, so you had a very good chance of getting addresses in low 2GB - not the case today.
- An extra instruction on a hot path mattered a lot more than how much it matters today.

We need to make sure that we have test coverage for all the variants with the software write watch. E.g. if you do not see the 32-bit constant ones getting hit in tests - we should not have them, so that they won't get hit by accident in production.